### PR TITLE
Fix or add more details about some MSTest rules

### PR DESCRIPTION
**Pull request by Evangelink**
---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/testing/mstest-analyzers/mstest0008.md](https://github.com/dotnet/docs/blob/8bc8e4a4595dea292a2772703ce4f017fe84732c/docs/core/testing/mstest-analyzers/mstest0008.md) | [MSTEST0008: TestInitialize method should have valid layout](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0008?branch=pr-en-us-41885) |
| [docs/core/testing/mstest-analyzers/mstest0009.md](https://github.com/dotnet/docs/blob/8bc8e4a4595dea292a2772703ce4f017fe84732c/docs/core/testing/mstest-analyzers/mstest0009.md) | ["MSTEST0009: TestCleanup method should have valid layout"](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0009?branch=pr-en-us-41885) |
| [docs/core/testing/mstest-analyzers/mstest0010.md](https://github.com/dotnet/docs/blob/8bc8e4a4595dea292a2772703ce4f017fe84732c/docs/core/testing/mstest-analyzers/mstest0010.md) | [MSTEST0010: ClassInitialize method should have valid layout](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0010?branch=pr-en-us-41885) |
| [docs/core/testing/mstest-analyzers/mstest0011.md](https://github.com/dotnet/docs/blob/8bc8e4a4595dea292a2772703ce4f017fe84732c/docs/core/testing/mstest-analyzers/mstest0011.md) | [docs/core/testing/mstest-analyzers/mstest0011](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0011?branch=pr-en-us-41885) |
| [docs/core/testing/mstest-analyzers/mstest0012.md](https://github.com/dotnet/docs/blob/8bc8e4a4595dea292a2772703ce4f017fe84732c/docs/core/testing/mstest-analyzers/mstest0012.md) | [MSTEST0012: AssemblyInitialize method should have valid layout](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0012?branch=pr-en-us-41885) |
| [docs/core/testing/mstest-analyzers/mstest0013.md](https://github.com/dotnet/docs/blob/8bc8e4a4595dea292a2772703ce4f017fe84732c/docs/core/testing/mstest-analyzers/mstest0013.md) | ["MSTEST0013: AssemblyCleanup method should have valid layout"](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0013?branch=pr-en-us-41885) |
| [docs/core/testing/mstest-analyzers/mstest0029.md](https://github.com/dotnet/docs/blob/8bc8e4a4595dea292a2772703ce4f017fe84732c/docs/core/testing/mstest-analyzers/mstest0029.md) | [MSTEST0029: Public method should be test method](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0029?branch=pr-en-us-41885) |
| [docs/core/testing/mstest-analyzers/mstest0030.md](https://github.com/dotnet/docs/blob/8bc8e4a4595dea292a2772703ce4f017fe84732c/docs/core/testing/mstest-analyzers/mstest0030.md) | [MSTEST0030: Type containing `[TestMethod]` should be marked with `[TestClass]`](https://review.learn.microsoft.com/en-us/dotnet/core/testing/mstest-analyzers/mstest0030?branch=pr-en-us-41885) |

### CODE REVIEW

**Comment by Evangelink** [comment]
Ready for review @IEvangelist

File: docs/core/testing/mstest-analyzers/mstest0008.md [hunk 1]
```diff
@@ -40,6 +40,13 @@ Methods marked with `[TestInitialize]` should follow the following layout to be
 - it should not take any parameter
 - return type should be `void`, `Task` or `ValueTask`
 
+The type declaring these methods should also respect the following rules:
+
+- the type should be a class
```
Comment (IEvangelist) [inline docs/core/testing/mstest-analyzers/mstest0008.md:45]:
```suggestion
- The type should be a `class`.
```

File: docs/core/testing/mstest-analyzers/mstest0008.md [hunk 2]
```diff
@@ -40,6 +40,13 @@ Methods marked with `[TestInitialize]` should follow the following layout to be
...
+- the class should be `public` or `internal` (if the test project is using the `[DiscoverInternals]` attribute)
```
Comment (IEvangelist) [inline docs/core/testing/mstest-analyzers/mstest0008.md:46]:
```suggestion
- The `class` should be `public` or `internal` (if the test project is using the `[DiscoverInternals]` attribute).
```

File: docs/core/testing/mstest-analyzers/mstest0008.md [hunk 3]
```diff
@@ -40,6 +40,13 @@ Methods marked with `[TestInitialize]` should follow the following layout to be
...
+- the class should not be `static`
```
Comment (IEvangelist) [inline docs/core/testing/mstest-analyzers/mstest0008.md:47]:
```suggestion
- The `class` shouldn't be `static`.
```

File: docs/core/testing/mstest-analyzers/mstest0008.md [hunk 4]
```diff
@@ -40,6 +40,13 @@ Methods marked with `[TestInitialize]` should follow the following layout to be
...
+- if the class is sealed, it should be marked with `[TestClass]` (or a derived attribute)
```
Comment (IEvangelist) [inline docs/core/testing/mstest-analyzers/mstest0008.md:48]:
```suggestion
- If the `class` is `sealed`, it should be marked with `[TestClass]` (or a derived attribute).
```

File: docs/core/testing/mstest-analyzers/mstest0008.md [hunk 5]
```diff
@@ -40,6 +40,13 @@ Methods marked with `[TestInitialize]` should follow the following layout to be
 - it should not take any parameter
 - return type should be `void`, `Task` or `ValueTask`
 
+The type declaring these methods should also respect the following rules:
```
Comment (IEvangelist) [inline docs/core/testing/mstest-analyzers/mstest0008.md:43]: Ah, I just noticed that these are all repeated. Let's practice a bit of DRY here and use an `include`. [We support include files](https://review.learn.microsoft.com/en-us/help/platform/markdown-reference?branch=main#included-markdown-files), in the _mstest-analyzers_ directory, create a new folder named _includes_ and add a new markdown file named _test-class-rules.md_.

```
└───📂 includes
     └─── test-class-rules.md
```

Then, update all of these MD files to use this format to include the newly created MD content:

```md
[!INCLUDE [test-class-rules](includes/test-class-rules.md)]
```

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**